utils/staking-miner/src/main.rs

```diff
@@ -0,0 +1,523 @@
+// that is not currently possible as each runtime has its unique `Call`, and all Calls are not
+// sharing any generic trait. In other words, to create the `UncheckedExtrinsic` of each chain, you
+// need the concrete `Call` of that chain as well.
+#[macro_export]
+macro_rules! any_runtime {
+	($($code:tt)*) => {
+		unsafe {
+			match $crate::RUNTIME {
+				$crate::AnyRuntime::Polkadot => {
+					use $crate::polkadot_runtime_exports::*;
+					$($code)*
+				},
+				$crate::AnyRuntime::Kusama => {
+					use $crate::kusama_runtime_exports::*;
+					$($code)*
+				},
+				$crate::AnyRuntime::Westend => {
+					use $crate::westend_runtime_exports::*;
+					$($code)*
+				}
+			}
+		}
+	}
+}
+
+#[derive(Debug, thiserror::Error)]
+enum Error {
+	Io(#[from] std::io::Error),
+	Jsonrpsee(#[from] jsonrpsee_ws_client::Error),
+	Codec(#[from] codec::Error),
+	Crypto(sp_core::crypto::SecretStringError),
+	RemoteExternalities(&'static str),
+	PalletMiner(EPM::unsigned::MinerError),
+	PalletElection(EPM::ElectionError),
+	PalletFeasibility(EPM::FeasibilityError),
+	AccountDoesNotExists,
+	IncorrectPhase,
+	AlreadySubmitted,
+}
+
+impl From<sp_core::crypto::SecretStringError> for Error {
+	fn from(e: sp_core::crypto::SecretStringError) -> Error {
+		Error::Crypto(e)
+	}
+}
+
+impl From<EPM::unsigned::MinerError> for Error {
+	fn from(e: EPM::unsigned::MinerError) -> Error {
+		Error::PalletMiner(e)
+	}
+}
+
+impl From<EPM::ElectionError> for Error {
+	fn from(e: EPM::ElectionError) -> Error {
+		Error::PalletElection(e)
+	}
+}
+
+impl From<EPM::FeasibilityError> for Error {
+	fn from(e: EPM::FeasibilityError) -> Error {
+		Error::PalletFeasibility(e)
+	}
+}
+
+impl std::fmt::Display for Error {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		<Error as std::fmt::Debug>::fmt(self, f)
+	}
+}
+
+#[derive(Debug, Clone, StructOpt)]
+enum Command {
+	/// Monitor for the phase being signed, then compute.
+	Monitor(MonitorConfig),
+	/// Just compute a solution now, and don't submit it.
+	DryRun(DryRunConfig),
+	/// Provide a solution that can be submitted to the chian as an emergency response.
+	EmergencySolution,
+}
+
+#[derive(Debug, Clone, StructOpt)]
+struct MonitorConfig {
+	/// They type of event to listen to.
+	///
+	/// Typically, finalized is safer and there is no chance of anything going wrong, but it can be
+	/// slower. It is recommended if the duration of the signed phase is longer than the a
+	#[structopt(long, default_value = "head", possible_values = &["head", "finalized"])]
+	listen: String,
+}
+
+#[derive(Debug, Clone, StructOpt)]
+struct DryRunConfig {
+	/// The block hash at which scraping happens. If none is provided, the latest head is used.
+	#[structopt(long)]
+	at: Option<Hash>,
+}
+
+#[derive(Debug, Clone, StructOpt)]
+struct SharedConfig {
+	/// The ws node to connect to.
+	#[structopt(long, default_value = DEFAULT_URI)]
+	uri: String,
+
+	/// The file from which we read the account seed.
```

Comment:
```suggestion
	/// The file from which we read the account seed. WARNING: use discretion when
	/// when choosing an account to submit solutions from.
```
Basically just an obvious warning don't like use a stash account or some account where you would be upset if all funds where lost, particularly since this is the first iteration for public consumption

Also maybe just a general "here be dragons" type warning for users would be beneficial for the community

Reply:
Great point, addressed.